src/windows/staff.c

```diff
@@ -1325,6 +1325,17 @@ void window_staff_viewport_init(rct_window* w){
 	window_invalidate(w);
 }
 
+void staff_get_valid_skins(int* ebx)
+{
+	init_scenery();
+	*ebx = 0;
```

Comment:
There is a difference, you are relying on the compiler optimising this which would only happen in Release mode if at all. Returning an integer will guarantee you a register return.